src/raftstore/store/peer_storage.rs

```diff
@@ -994,54 +1032,79 @@ impl PeerStorage {
 }
 
 /// Delete all meta belong to the region. Results are stored in `wb`.
-pub fn clear_meta(engine: &DB,
-                  wb: &WriteBatch,
+pub fn clear_meta(wb: &WriteBatch,
+                  raft_wb: &WriteBatch,
+                  raft_engine: &DB,
                   region_id: u64,
-                  raft_state: &RaftLocalState)
+                  raft_state: &RaftLocalState,
+                  is_recovery: bool)
```

Comment:
if restart happen when we write region_state to tombstone, but not clear raft logs.   the clear_stale_meta need remove all raft logs.
is_recovery = false is for normal clear_meta. https://github.com/pingcap/tikv/pull/2095